client/me/purchases/use-field-definitions.ts

```diff
@@ -0,0 +1,46 @@
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import { getTransactionFieldDefinitions } from './billing-history/field-definitions';
+import type { Purchases } from '@automattic/data-stores';
+import type { BillingTransaction } from 'calypso/state/billing-transactions/types';
+
+function getFieldDefinitionsByType( {
+	itemType,
+	items,
+	translate
+}: {
+	itemType: string | null;
+	items: BillingTransaction[] | Purchases.Purchase[] | null;
+	translate: typeof useTranslate;
+} ) {
+	if ( 'null' !== itemType || ! items ) {
```

Comment:
Because you have an early return here if `itemType` is _not_ the string `'null'`, none of the other code below here that checks for other item types will fire. Did you mean to check to see if `itemType` is `'null'` instead?